more_itertools/more.py

```diff
@@ -74,6 +74,32 @@
 _marker = object()
 
 
+def _decorator_factory(cls):
```

Comment:
Trying to think of a name for this that allows for it returning a non-decorator also (as it does). `_either_decorator_or_wrapper`? `_optionally_a_decorator`?

Reply:
I went with `_wrapper_factory`, since that covers both - it's either wrapping the iterable or the function.